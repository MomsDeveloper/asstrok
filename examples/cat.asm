INT:
    OUT R2
    ST R2, 0
    RST

START:
    LD R1, 0
    JE R1, END
    JMP START

END:
    HLT
