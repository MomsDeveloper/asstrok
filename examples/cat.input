[(0, 'c'), (10, 'a'), (20, 't'), (30, '\0')]
