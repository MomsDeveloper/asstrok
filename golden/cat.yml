in_source: |-
  INT:
      OUT R2
      ST R2, 0
      RST

  START:
      LD R1, 0
      JE R1, END
      JMP START

  END:
      HLT
in_stdin: |-
  [(0, 'c'), (10, 'a'), (20, 't'), (30, '\0')]
out_log: |
  INFO     root:machine.py:38 Starting simulation
  INFO     root:machine.py:39 Current State:TICK: 0	PC: 4
  ALU_L: 0	ALU_R: 0	ALU_OUT: 0
  R1: 0	R2: 0	SP: 255
  STACK: []
  MEMORY: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IOMemoryInstructionImm(opcode=<Opcode.LD: 12>, dest=<Registers.R1: 0>, src=0, arg_type=<ArgType.IMM: 1>)
  INFO     root:machine.py:40 
  INFO     root:machine.py:54 Current State:Interrupted with input: c
  TICK: 1	PC: 5
  ALU_L: 0	ALU_R: 0	ALU_OUT: 0
  R1: 0	R2: 0	SP: 255
  STACK: []
  MEMORY: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]

  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 7	PC: 1
  ALU_L: 0	ALU_R: 0	ALU_OUT: 0
  R1: 0	R2: 99	SP: 251
  STACK: [0, 5, 0, 0]
  MEMORY: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IOOutInstruction(opcode=<Opcode.OUT: 14>, src=<Registers.R2: 1>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 8	PC: 2
  ALU_L: 99	ALU_R: 0	ALU_OUT: 99
  R1: 0	R2: 99	SP: 251
  STACK: [0, 5, 0, 0]
  MEMORY: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IOMemoryInstructionImm(opcode=<Opcode.ST: 13>, dest=<Registers.R2: 1>, src=0, arg_type=<ArgType.IMM: 1>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 9	PC: 3
  ALU_L: 99	ALU_R: 0	ALU_OUT: 99
  R1: 0	R2: 99	SP: 251
  STACK: [0, 5, 0, 0]
  MEMORY: [99, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IORstInstruction(opcode=<Opcode.RST: 15>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 12	PC: 5
  ALU_L: 5	ALU_R: 0	ALU_OUT: 5
  R1: 0	R2: 0	SP: 255
  STACK: []
  MEMORY: [99, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  JumpEqInstruction(opcode=<Opcode.JE: 9>, src=<Registers.R1: 0>, addr=7)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:Interrupted with input: a
  TICK: 13	PC: 7
  ALU_L: 5	ALU_R: 0	ALU_OUT: 5
  R1: 0	R2: 0	SP: 255
  STACK: []
  MEMORY: [99, 0, 0, 0, 0, 0, 0, 0, 0, 0]

  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 19	PC: 1
  ALU_L: 0	ALU_R: 0	ALU_OUT: 0
  R1: 0	R2: 97	SP: 251
  STACK: [0, 7, 0, 0]
  MEMORY: [99, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IOOutInstruction(opcode=<Opcode.OUT: 14>, src=<Registers.R2: 1>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 20	PC: 2
  ALU_L: 97	ALU_R: 0	ALU_OUT: 97
  R1: 0	R2: 97	SP: 251
  STACK: [0, 7, 0, 0]
  MEMORY: [99, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IOMemoryInstructionImm(opcode=<Opcode.ST: 13>, dest=<Registers.R2: 1>, src=0, arg_type=<ArgType.IMM: 1>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:Interrupted with input: t
  TICK: 21	PC: 3
  ALU_L: 97	ALU_R: 0	ALU_OUT: 97
  R1: 0	R2: 97	SP: 251
  STACK: [0, 7, 0, 0]
  MEMORY: [97, 0, 0, 0, 0, 0, 0, 0, 0, 0]

  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 27	PC: 1
  ALU_L: 0	ALU_R: 0	ALU_OUT: 0
  R1: 0	R2: 116	SP: 247
  STACK: [0, 3, 97, 0, 0, 7, 0, 0]
  MEMORY: [97, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IOOutInstruction(opcode=<Opcode.OUT: 14>, src=<Registers.R2: 1>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 28	PC: 2
  ALU_L: 116	ALU_R: 0	ALU_OUT: 116
  R1: 0	R2: 116	SP: 247
  STACK: [0, 3, 97, 0, 0, 7, 0, 0]
  MEMORY: [97, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IOMemoryInstructionImm(opcode=<Opcode.ST: 13>, dest=<Registers.R2: 1>, src=0, arg_type=<ArgType.IMM: 1>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 29	PC: 3
  ALU_L: 116	ALU_R: 0	ALU_OUT: 116
  R1: 0	R2: 116	SP: 247
  STACK: [0, 3, 97, 0, 0, 7, 0, 0]
  MEMORY: [116, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IORstInstruction(opcode=<Opcode.RST: 15>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 32	PC: 3
  ALU_L: 3	ALU_R: 0	ALU_OUT: 3
  R1: 0	R2: 97	SP: 251
  STACK: [0, 7, 0, 0]
  MEMORY: [116, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IORstInstruction(opcode=<Opcode.RST: 15>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:Interrupted 
  TICK: 35	PC: 7
  ALU_L: 7	ALU_R: 0	ALU_OUT: 7
  R1: 0	R2: 0	SP: 255
  STACK: []
  MEMORY: [116, 0, 0, 0, 0, 0, 0, 0, 0, 0]

  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 41	PC: 1
  ALU_L: 0	ALU_R: 0	ALU_OUT: 0
  R1: 0	R2: 0	SP: 251
  STACK: [0, 7, 0, 0]
  MEMORY: [116, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IOOutInstruction(opcode=<Opcode.OUT: 14>, src=<Registers.R2: 1>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 42	PC: 2
  ALU_L: 0	ALU_R: 0	ALU_OUT: 0
  R1: 0	R2: 0	SP: 251
  STACK: [0, 7, 0, 0]
  MEMORY: [116, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IOMemoryInstructionImm(opcode=<Opcode.ST: 13>, dest=<Registers.R2: 1>, src=0, arg_type=<ArgType.IMM: 1>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 43	PC: 3
  ALU_L: 0	ALU_R: 0	ALU_OUT: 0
  R1: 0	R2: 0	SP: 251
  STACK: [0, 7, 0, 0]
  MEMORY: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  IORstInstruction(opcode=<Opcode.RST: 15>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:54 Current State:TICK: 46	PC: 7
  ALU_L: 7	ALU_R: 0	ALU_OUT: 7
  R1: 0	R2: 0	SP: 255
  STACK: []
  MEMORY: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
  Instruction to execute:
  ManagementInstruction(opcode=<Opcode.HLT: 0>)
  INFO     root:machine.py:55 
  INFO     root:machine.py:78 Simulation finished
  INFO     root:machine.py:79 Output buffer:
out_stdout: |
  ============================================================
  ['c', 'a', 't', '\x00']
out_code: !!binary |
  AASgAegA3ADwAMgAkAeABAAA
